Color variables for scarves/beanie fix

Niknakflak opened the pull request:
- removes minor iconbloat from scarves and colors them via the variables
- minor beanie fixes/organization
- Code by JJRcop: "Mob overlays for ties on humans and corgis now inherit color and alpha to make this functional"
- Turns like 40 icons into 3

Niknakflak committed:
Does some stuff

Commit:
Fixes ties on humans and corgis not applying color and alpha to the overlays

Niknakflak committed:
Merge pull request #4 from JJRcop/TieOverlayFix

Fixes tie mob overlays for corgis and humans

Cheridan commented:
awesome

Cheridan commented on code/datums/dog_fashion.dm:
```diff
@@ -10,6 +10,8 @@
 	var/icon_file
 	var/icon_state
+	var/alpha
+	var/color
```
imo change these var names to prevent confusion with inbuilt atom alpha and color vars.

Iamgoofball replied:
actually how does this compile BYOND should be throwing a fit due to reserved var names

Niknakflak replied:
JRrcop code, he said he tested, I dunno.

JJRcop replied:
it works you know, i don't play around with my testing. but @Iamgoofball it works for the same reason icon_state works, this is a datum.

Iamgoofball replied:
h u h

Niknakflak replied:
so do we have to change them

JJRcop replied:
Niknakflak#5

Commit:
Renames /datum/dog_fashion vars to reduce confusion
Also fixes issue introduced in d2fd3f36918b9abfa833b75fda8af3c124c85336 where custom DF icon_states (now obj_icon_state) would not carry over to the corgi

Niknakflak committed:
Merge pull request #5 from JJRcop/grublebrumdle

Renames /datum/dog_fashion vars to reduce confusion

Commit:
Fixes consistency mess up with dog_fashion
caused by 2595ce2d322cd74653eca2be43f79ad63104e61a

Niknakflak committed:
Merge pull request #6 from JJRcop/grublebrumdle

Fixes consistency mess up with dog_fashion

JJRcop commented:
sorry

Niknakflak commented:
squishy

AnturK commented on code/modules/clothing/under/ties.dm:
```diff
@@ -214,55 +214,55 @@
-	icon_state = "orangescarf"
-	item_color = "orangescarf"
+	icon_state = "scarf"
+	color = "#C67A4B" //orange
```
This would be funnier if purple was capitalised.
